rackcity/utils/change_planner_utils.py

```diff
@@ -147,16 +151,24 @@ def get_change_plan(change_plan_id):
 
 def get_racked_assets_for_cp(change_plan):
     racked_assets = Asset.objects.filter(offline_storage_site__isnull=True)
-    racked_assets_cp = AssetCP.objects.filter(change_plan=change_plan, offline_storage_site__isnull=True)
+    racked_assets_cp = AssetCP.objects.filter(
+        change_plan=change_plan, offline_storage_site__isnull=True
+    )
     for asset_cp in racked_assets_cp:
         if asset_cp.related_asset and (asset_cp.related_asset in racked_assets):
             racked_assets = racked_assets.filter(~Q(id=asset_cp.related_asset.id))
+    ## don't return decommissioned asset_cps:
+    racked_assets_cp = racked_assets_cp.filter(is_decommissioned=False)
     return racked_assets, racked_assets_cp
 
 
 def get_offline_storage_assets_for_cp(change_plan):
     stored_assets = Asset.objects.filter(offline_storage_site__isnull=False)
-    stored_assets_cp = AssetCP.objects.filter(change_plan=change_plan, offline_storage_site__isnull=False)
+    stored_assets_cp = AssetCP.objects.filter(
```

Comment:
ignore decomissioned assets

Reply:
i think u wanna do the same thing here that u did above in `get_racked_assets_for_cp` and filter out decommissioned AssetCP's AFTER filtering out all change-plannified stored Asset's. cuz if u have stored Asset live, and u have decommissioned it in a cp so its a stored + decommissioned AssetCP, then u DON'T want to return that Asset in the query so u need the related AssetCP here. but u DONT want to return the AssetCP either so u have to filter it out after

Reply:
good point